### PR TITLE
Remove support for passing strategy strings to accelerator

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of  #12521

### Does your PR introduce any breaking changes? If yes, please list them.

Yes. This is no longer supported:

```py
Trainer(accelerator="ddp_spawn")
Trainer(accelerator="deepspeed")
Trainer(accelerator="ddp_cpu")
```

Instead, use the strategy argument:
```py
Trainer(strategy="ddp_spawn")
Trainer(strategy="deepspeed")
Trainer(strategy="ddp", accelerator="cpu")
```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @kaushikb11 @awaelchli @ninginthecloud

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -99,6 +99,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed deprecated `GPUStatsMonitor` callback ([#12554](https://github.com/PyTorchLightning/pytorch-lightning/pull/12554))
 
 
+- Removed support for passing strategy names to the accelerator Trainer argument ([#12696](https://github.com/PyTorchLightning/pytorch-lightning/pull/12696))
+
+
```
Comment (krshrimali) [inline CHANGELOG.md:104]:
```suggestion
- Removed support for passing strategy names to the accelerator Trainer argument ([#12696](https://github.com/PyTorchLightning/pytorch-lightning/pull/12696))


- Removed support for passing strategy class to the accelerator Trainer argument ([#12696](https://github.com/PyTorchLightning/pytorch-lightning/pull/12696))


```

File: tests/accelerators/test_accelerator_connector.py [hunk 2]
```diff
@@ -681,7 +430,7 @@ def test_strategy_choice_ddp_spawn_cpu(tmpdir):
 @mock.patch.dict(os.environ, {"CUDA_VISIBLE_DEVICES": "0,1"})
 @mock.patch("torch.cuda.device_count", return_value=2)
 @mock.patch("torch.cuda.is_available", return_value=True)
-def test_strategy_choice_ddp(cuda_available_mock, device_count_mock):
+def test_strategy_choice_ddp(*_):
```
Comment (krshrimali) [inline tests/accelerators/test_accelerator_connector.py:433]: Nice! I didn't know about this.

**Review by krshrimali — APPROVED** [review]
Thanks, @awaelchli! 🚀 LGTM, just a minor comment.

**Review by justusschock — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/utilities/enums.py [hunk 3]
```diff
@@ -240,7 +239,6 @@ class _StrategyType(LightningEnum):
     DP = "dp"
     DDP = "ddp"
     DDP2 = "ddp2"
-    DDP_CPU = "ddp_cpu"
     DDP_SPAWN = "ddp_spawn"
     TPU_SPAWN = "tpu_spawn"
```
Comment (carmocca) [inline pytorch_lightning/utilities/enums.py:243]: shouldn't we also remove this? in a follow-up?

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 4]
```diff
@@ -286,14 +271,6 @@ def _check_config_and_set_final_flags(
         if accelerator is not None:
             if accelerator in self._accelerator_types or accelerator == "auto" or isinstance(accelerator, Accelerator):
                 self._accelerator_flag = accelerator
-            elif accelerator in self._registered_strategies or isinstance(accelerator, Strategy):
```
Comment (four4fish) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:289]: Should we add a else to throw an exception here if user passed in an unrecognizable accelerator?

File: pytorch_lightning/utilities/enums.py (hunk 3 above)
Comment (four4fish) [reply pytorch_lightning/utilities/enums.py:243]: ddp_cpu could still be passed in throw strategy flag https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/accelerator_connector.py#L251-L255
This PR only remove support of passing in by accelerator flag

Comment (carmocca) [reply pytorch_lightning/utilities/enums.py:243]: I was asking about "tpu_spawn"

File: pytorch_lightning/trainer/connectors/accelerator_connector.py (hunk 4 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:289]: yes and no. That would need more work and I think outside of scope here. see #12698

File: pytorch_lightning/utilities/enums.py (hunk 3 above)
Comment (awaelchli) [reply pytorch_lightning/utilities/enums.py:243]: yeah we could remove that. 
@four4fish soon we should probably also remove that error message specific to ddp_cpu you pointed out, as it will just be an invalid strategy name like any other.

Comment (awaelchli) [reply pytorch_lightning/utilities/enums.py:243]: Let's do that in a follow-up. TPU_SPAWN is used in the `interactive_compatible` error messaging methods and I don't feel comfortable changing the behavior in this PR.
